internal/resolver/delegatingresolver/delegatingresolver.go

```diff
@@ -0,0 +1,196 @@
+// resolvers to resolver targetURI as well as proxy address.
+
+package delegatingresolver
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+	"net/url"
+	"sync"
+
+	"google.golang.org/grpc/attributes"
+	"google.golang.org/grpc/resolver"
+	"google.golang.org/grpc/serviceconfig"
+)
+
+type delegatingResolver struct {
+	target         resolver.Target
+	cc             resolver.ClientConn
+	isProxy        bool
+	targetResolver resolver.Resolver
+	proxyResolver  resolver.Resolver
+	targetAddrs    []resolver.Address
+	proxyAddrs     []resolver.Address
+	ctx            context.Context
+	cancel         context.CancelFunc
+	mu             sync.Mutex // Protects the state below
+
+}
+
+type innerClientConn struct {
+	*delegatingResolver // might not be needed
+	resolverType        string
+}
+
+var (
+	// The following variable will be overwritten in the tests.
+	httpProxyFromEnvironment = http.ProxyFromEnvironment
+)
+
+func mapAddress(address string) (*url.URL, error) {
+	req := &http.Request{
+		URL: &url.URL{
+			Scheme: "https",
+			Host:   address,
+		},
+	}
+	url, err := httpProxyFromEnvironment(req)
+	if err != nil {
+		return nil, err
+	}
+	return url, nil
+}
+
+func New(target resolver.Target, cc resolver.ClientConn, opts resolver.BuildOptions, targetResolverBuilder resolver.Builder) (resolver.Resolver, error) {
+	r := &delegatingResolver{
+		target: target,
+		cc:     cc,
+	}
+	proxyURL, err := mapAddress(target.Endpoint())
+	if err != nil {
+		return nil, err
+	}
+	r.isProxy = true
+	if proxyURL == nil {
+		r.isProxy = false
+	}
+	if r.isProxy {
+		if target.URL.Scheme == "dns" {
+			r.targetAddrs = []resolver.Address{{Addr: r.target.Endpoint()}}
+		} else {
+			r.targetResolver, err = targetURIResolver(target, opts, targetResolverBuilder, r)
+		}
+		r.proxyResolver, err = proxyURIResolver(proxyURL, opts, r)
+	} else {
+		// r.targetResolver, err = targetURIResolver(target, opts, targetResolverBuilder, r) //
+		return targetResolverBuilder.Build(target, cc, opts)
+	}
+	if err != nil {
+		return nil, err
+	}
+	return r, nil
+}
+
+func targetURIResolver(target resolver.Target, opts resolver.BuildOptions, targetResolverBuilder resolver.Builder, resolver *delegatingResolver) (resolver.Resolver, error) {
+	targetBuilder := targetResolverBuilder
+	fmt.Printf("the delegated resolver buildr is %v \n", targetBuilder)
+	if targetBuilder == nil {
+		return nil, fmt.Errorf("resolver for target scheme %q not found", target.URL.Scheme)
+	}
+	fmt.Printf("the delegated resolver buildr 1 is %v \n", targetBuilder)
+	return targetBuilder.Build(target, &innerClientConn{resolver, "target"}, opts)
+}
+
+func proxyURIResolver(proxyURL *url.URL, opts resolver.BuildOptions, presolver *delegatingResolver) (resolver.Resolver, error) {
+	scheme := "dns"
+	proxyBuilder := resolver.Get(scheme)
+	if proxyBuilder == nil {
+		return nil, fmt.Errorf("resolver for proxy not found")
+	}
+	host := proxyURL.Hostname()
```

Comment:
I earlier had `proxyTarget:=resolver.Target{URL: *proxyURL}` but that resulted in empty `url.Path` which in turn resulted in dns returning error saying missing endpoint. I am not sure why this is happening, form what I can see proxyURL is of the form `http://localhost:8000` like this and it gets stored in `URL.Host = localhost:8000` and `URL.path` remains empty. This way seems to solve the problem as we use `URL.parse` on the host which saves it in path also returning desired values. I have also confirmed it using the route_guide example.